### ISSUE TITLE
proposal for making etcd scheduleCompact configurable

### ISSUE THREAD

**Issue by WIZARD-CXY** [etcd-io/etcd#10482]
Now the etcd use `scheduleCompaction` in https://github.com/etcd-io/etcd/blob/master/mvcc/kvstore_compaction.go#L33 do the scheduler compaction.
But the batchsize and compact interval is fixed as below:
 `batchsize := int64(10000)` and `case <-time.After(100 * time.Millisecond)`.
 I think we'd better make them configurable.
One use case: in large clusters we can make  compaction in the background more gently by configure batchsize  smaller.

**Comment by WIZARD-CXY** [comment]
@xiang90 @hexfusion WDYT

**Comment by WIZARD-CXY** [comment]
Maybe we can add two field in the `store` object like `compactBatchSize` and `compactInterval` and use `NewStore` to init them?

**Comment by jingyih** [comment]
I think this makes sense. Also, I noticed there is no config struct for store. My suggestion is, if you are passing these fields when creating new store, let's use a config struct. So that later if we decided to add more field, we do not change the function signature of `newStore`.

In case I was not clear, I mean something similar to this:
https://github.com/etcd-io/etcd/blob/8c228d692bfa516e1c0977922f061b9a0fb1ae0f/mvcc/backend/backend.go#L106
